src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/CompareAttribute.cs

```diff
@@ -56,35 +56,23 @@ protected override ValidationResult IsValid(object value, ValidationContext vali
             {
                 if (OtherPropertyDisplayName == null)
                 {
-                    OtherPropertyDisplayName = GetDisplayNameForProperty(validationContext.ObjectType, OtherProperty);
+                    OtherPropertyDisplayName = GetDisplayNameForProperty(validationContext.ObjectType, otherPropertyInfo);
                 }
                 return new ValidationResult(FormatErrorMessage(validationContext.DisplayName));
             }
             return null;
         }
 
-        private static string GetDisplayNameForProperty(Type containerType, string propertyName)
+        private string GetDisplayNameForProperty(Type containerType, PropertyInfo property)
         {
-            var property = containerType.GetRuntimeProperties()
-                .SingleOrDefault(
-                    prop =>
-                        IsPublic(prop) &&
-                        string.Equals(propertyName, prop.Name, StringComparison.OrdinalIgnoreCase));
-
-            if (property == null)
-            {
-                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture,
-                    SR.Common_PropertyNotFound, containerType.FullName, propertyName));
-            }
-
             var attributes = CustomAttributeExtensions.GetCustomAttributes(property, true);
             var display = attributes.OfType<DisplayAttribute>().FirstOrDefault();
             if (display != null)
             {
                 return display.GetName();
             }
 
-            return propertyName;
+            return OtherProperty;
```

Comment:
The old value passed to the parameter `propertyName` was `OtherProperty`. Now that that parameter is gone, we use `OtherProperty` directly